bittensor/utils/__init__.py

```diff
@@ -115,126 +237,155 @@ def solve_for_difficulty_fast( subtensor, wallet, num_processes: int = None, upd
         update_interval: int
             Number of nonces to solve before updating block information.
     Note: 
-    - We should modify the number of processes based on user input.
     - We can also modify the update interval to do smaller blocks of work,
         while still updating the block information after a different number of nonces,
         to increase the transparency of the process while still keeping the speed.
     """
     if num_processes == None:
-        num_processes = multiprocessing.cpu_count()
+        # get the number of allowed processes for this process
+        num_processes = len(os.sched_getaffinity(0))
+
+    if update_interval is None:
+        update_interval = 50_000
         
+    limit = int(math.pow(2,256)) - 1
+
+    console = bittensor.__console__
+    status = console.status("Solving")
+
+    best_seal: bytes
+    best_number: int
+    best_number = float('inf')
+
+    curr_block = multiprocessing.Array('h', 64, lock=True) # byte array
+    curr_block_num = multiprocessing.Value('i', 0, lock=True) # int
+    curr_diff = multiprocessing.Array('Q', [0, 0], lock=True) # [high, low]
+
+    def update_curr_block(block_number: int, block_bytes: bytes, diff: int, lock: multiprocessing.Lock):
+        with lock:
+            curr_block_num.value = block_number
+            for i in range(64):
+                curr_block[i] = block_bytes[i]
+            curr_diff[0] = diff >> 32
+            curr_diff[1] = diff & 0xFFFFFFFF # low 32 bits
+
+    status.start()
+
+    # Establish communication queues
+    stopEvent = multiprocessing.Event()
+    stopEvent.clear()
+    best_queue = multiprocessing.Queue()
+    solution_queue = multiprocessing.Queue()
+    time_queue = multiprocessing.Queue()
+    check_block = multiprocessing.Lock()
+    
+    # Start consumers
+    solvers = [ Solver(i, num_processes, update_interval, best_queue, time_queue, solution_queue, stopEvent, curr_block, curr_block_num, curr_diff, check_block, limit)
```

Comment:
I see... so all of these objects are shared across all of the processes and each process has a random starting point?